src/overlays/actors/ovl_En_Bal/z_en_bal.c

```diff
@@ -64,108 +149,1026 @@ static DamageTable D_80A63CE0 = {
+                       TINGLE_LIMB_MAX);
+    if (gSaveContext.save.saveInfo.playerData.isMagicAcquired) {
+        Animation_Change(&this->skelAnime, &gTingleTalkIdleAnim, 1.0f, 0.0f, lastFrame, ANIMMODE_LOOP, -10.0f);
+    }
+    ActorShape_Init(&this->picto.actor.shape, 0.0f, ActorShadow_DrawCircle, 0.0f);
+    this->skyFloatPhase = 0;
+    this->prevTextId = 0;
+    this->prevBalloonAction = EN_BAL_BALACT_NONE;
+    this->timer = 0;
+    this->watchTarget = EN_BAL_WATCHTRG_NONE;
+    this->inflateEarly = false;
+    this->isTalking = false;
+    this->eyeTexIndex = EN_BAL_EYETEX_OPEN;
+    this->eyeTimer = 60;
+    this->forceEyesShut = false;
+    Collider_InitJntSph(play, &this->collider);
+    Collider_SetJntSph(play, &this->collider, &this->picto.actor, &sJntSphInit, &this->colliderElement);
+    CollisionCheck_SetInfo2(&this->picto.actor.colChkInfo, &sDamageTable, &sColChkInfoInit);
+    Actor_UpdateBgCheckInfo(play, &this->picto.actor, 0.0f, 0.0f, 0.0f, UPDBGCHECKINFO_FLAG_4);
+    this->picto.validationFunc = EnBal_ValidatePictograph;
+    if (!gSaveContext.save.saveInfo.playerData.isMagicAcquired) {
+        this->picto.actor.world.pos.y = this->picto.actor.floorHeight;
+        EnBal_SetMainColliderToHead(this);
+        EnBal_SetupGroundIdle(this);
+    } else {
+        EnBal_SetupFloatIdle(this);
+    }
+}
+
+void EnBal_Destroy(Actor* thisx, PlayState* play) {
+    EnBal* this = THIS;
+
+    Collider_InitJntSph(play, &this->collider);
+}
+
+void EnBal_SetMainColliderToBalloon(EnBal* this) {
+    this->collider.elements->dim.limb = TINGLE_LIMB_BALLOON;
+    this->collider.elements->dim.modelSphere.radius = 40;
+    this->collider.elements->dim.modelSphere.center.x = 2200;
+}
+
+void EnBal_SetMainColliderToHead(EnBal* this) {
+    this->collider.elements->dim.limb = TINGLE_LIMB_HEAD;
+    this->collider.elements->dim.modelSphere.radius = 25;
+    this->collider.elements->dim.modelSphere.center.x = 0;
+}
+
+s32 EnBal_ValidatePictograph(PlayState* play, Actor* thisx) {
+    s32 pictoValid;
+    EnBal* this = THIS;
+
+    pictoValid = Snap_ValidatePictograph(play, &this->picto.actor, PICTO_VALID_TINGLE, &this->picto.actor.focus.pos,
+                                         &this->picto.actor.shape.rot, 10.0f, 400.0f, 0x4000);
+    if (!pictoValid) {
+        this->forceEyesShut = true;
+        this->idleAnimStage = EN_BAL_IDLESTAGE_WAIT;
+    }
+    return pictoValid;
+}
+
+void EnBal_SetupFloatIdle(EnBal* this) {
+    this->timer = 0;
+    EnBal_SetMainColliderToBalloon(this);
+    Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimationInfo, TINGLE_ANIM_FLOAT_DRAW);
+
+    this->picto.actor.focus.pos = this->picto.actor.world.pos;
+    this->picto.actor.focus.pos.y = this->picto.actor.world.pos.y + 100.0f;
+
+    this->actionFunc = EnBal_FloatIdle;
+}
+
+void EnBal_FloatIdle(EnBal* this, PlayState* play) {
+    f32 scaleFactor;
+
+    this->skyFloatPhase += 800;
```

Comment:
```suggestion
    this->skyFloatPhase += 0x320;
```
We always try to keep rotations/angles/phases in hex for now